tests/misc_test.go

```diff
@@ -371,6 +373,52 @@ var _ = Describe("RHEL VM creation", func() {
 	)
 })
 
+var _ = Describe("SSP Status Phase", func() {
+	BeforeEach(func() {
+		waitUntilDeployed()
+	})
+
+	It("should set phase to Deploying when owned resource gets deletionTimestamp", func() {
+		finalizerName := "ssp.kubernetes.io/deletion-test"
+
+		configMapRes := testResource{
+			Name:      validator.ConfigMapName,
+			Namespace: strategy.GetNamespace(),
+			Resource:  &core.ConfigMap{},
+		}
+
+		defer func() {
+			removeFinalizer(configMapRes, finalizerName)
+		}()
+
+		addFinalizer(configMapRes, finalizerName)
```

Comment:
The usual pattern is to swap these calls. A call to `defer` is done after the object is created. `addFinalizer` could fail and panic, and then there is no reason to call `removeFinalizer`.